libraries/rush-sdk/src/index.ts

```diff
@@ -37,18 +37,14 @@ if (rushLibModule === undefined) {
         // Try to resolve rush-lib from the caller's folder
         try {
           const rushLibModulePath: string = Import.resolveModule({
-            modulePath: '@microsoft/rush-lib',
+            modulePath: RUSH_LIB_NAME,
             baseFolderPath: callerPackageFolder
           });
 
           rushLibModule = require(rushLibModulePath);
         } catch (error) {
           // If we fail to resolve it, ignore the error
         }
-
-        if (rushLibModule !== undefined) {
-          global.___rush___rushLibModule = rushLibModule;
```

Comment:
Add a note about why this isn't being set by the "`// SCENARIO 2:  ...`" comment.

Reply:
There's an interesting question of whether it would EVER be valid to load two instances of `rush-lib` simultaneously.

For example, suppose a plugin imports some library that has its own bundled `rush-lib` (scenario 1+2).  Or suppose that a tool built with `rush-lib` imports another tool that also has its own `rush-lib` (scenario 2+2). 

- With the code snippet above, whoever loads `rush-lib` first would win.  
- Whereas with the code snippet deleted, then we load two copies side-by-side.

Reply:
Maybe the safest policy is:
- If `rushLibModule !== undefined` then simply report an error.  If that error is blocking a valid use case, let them file a bug and explain it to us.  :-)

Reply:
I think we should just throw an error.

Reply:
I thought about this some more.  Imagine this pretty reasonable seeming case:

- `my-tool` depends on `my-library`, and provides `"rush-lib"` as a regular dependency
- `my-tool` imports `rush-sdk`
- `my-library` also imports `rush-sdk`, and possibly installs `"rush-lib"` as a dev dependency for its Jest tests

In this situation, it seems like we can't do much better than "whoever loads `rush-lib` first decides the instance."

To allow that, I've reverted my change above and added some code comments.  When we implement Scenario 3 we can add more validation.